### PR TITLE
Check for .json file extension before JSON parse. Was throwing when r…

### PR DESCRIPTION
**Pull request by msand**
…eading js files. Related to #597

### CODE REVIEW

File: server/read-page.js [hunk 1]
```diff
@@ -12,11 +12,15 @@ async function readPage (path) {
     return cache[f]
   }
 
-  const source = await fs.readFile(f, 'utf8')
-  const { component } = JSON.parse(source)
+  let content = await fs.readFile(f, 'utf8')
 
-  cache[f] = component
-  return component
+  if (f.slice(-5) === '.json') {
```
Comment (nkzawa) [inline server/read-page.js]: I think this is always `.json` ? @arunoda

Comment (arunoda) [inline server/read-page.js]: We need to have `.js` as well. In order to support `dist`. Loading SSR files.

Comment (arunoda) [inline server/read-page.js]: Oh! wait. But for here, it's always `.json`. 
If there's no `.json` file, we need to throw an error.
Then I think we don't need a change here.

**Comment by ryancole** [comment]
Just tested `msand:fix-windows-error`, and it does indeed cause the error to go away. I'm on Windows 10.

File: server/read-page.js [hunk 2]
```diff
@@ -12,11 +12,15 @@ async function readPage (path) {
     return cache[f]
   }
 
-  const source = await fs.readFile(f, 'utf8')
-  const { component } = JSON.parse(source)
+  let content = await fs.readFile(f, 'utf8')
```
Comment (arunoda) [inline server/read-page.js]: @msand Would you mind removing all the changes done to this page.
We always need to read `.json` files so if there's no `.json` file, that's an error.

**Review by arunoda — CHANGES_REQUESTED** [review]

**Comment by arunoda** [comment]
Took this PR with https://github.com/zeit/next.js/pull/611

File: server/read-page.js (hunk 2 above)
Comment (msand) [inline server/read-page.js]: Alright, I figured as much, should've made a separate pr to begin with. Thanks

**Pull request closed without merging**